README.md

```diff
@@ -3,129 +3,92 @@
+    - Provide API access KEY and APP ID enrollment APIs.
+    - Manage key properties, store/retrieve those info to/from DB. (WIP)
+        - Storing CMK Keyblobs, Key Versions and CMK rotation management.
+        - KeyId mapping, Alias name, Origin, KeySpec, Key usage.
+---
 
-### ehsm-dkeyserver
-It will be deployed in an HSM-equipped platform or an SGX-capable platform, and runs as a TCP/IP socket server.
+## APPID and APIKey Enrollment
+Since only the user with valid APPID and APIKey could request the public cryptographic restful APIs, eHSM-KMS provides a new Enroll APP which is used to retrieve the APPID and APIKey from the eHSM-core enclave via the remote secure channel (based on the SGX remote attestation).
 
-- Responsible for generating DK(domainkey) and protecting it in the HSM or enclave.
+The below diagram depicts the flow how to enroll a valid APPID and APIKey for the user,
+![apikey-and-appid-enrollment](./docs/diagrams/apikey-and-appid-enrollment.PNG)
 
-- Respond to the connection request from ehsm-dkeycache agent and deploy the DK to it via a remote secure channel
-based on the SGX remote attestation.
+For more details about each crypto APIs, please visit the doc [API_Reference.md](./docs/API_Reference.md).
 
-- It may also provide some eHSM Configuration interfaces to the Customer Operator Panel to manage the domains.
-(depending on the requirements, and not implemented yet.)
+---
 
+## Introduction to Keys used in eHSM-KMS
 
-### ehsm-dkeycache
-It will be deployed in the SGX worker node(1:1 map with ehsm-core in a Pod), and runs as a proxy agent service.
+eHSM-KMS will provide different kinds of keys, but none of them will be exposed outside of the SGX enclave.
+The below diagram depicts the overview of the keys used in the eHSM-KMS,
+![keys-in-ehsm-kms](./docs/diagrams/keys-in-ehsm-kms.PNG)
 
-- Responsible for retrieving the DK from the ehsm-dkeyserver via remote secure channel and then distributing to
-ehsm-core instance via local secure channel.
+ - DK (DomainKey) is an AES key that is used to protect/wrap all the CMKs in a domain (region).
+    - It must be securely distributed/provisioned cross all SGX-capable machines (or nodes) in a domain.
+ - CMK (Customer Master Key) could be symmetric or asymmetric keys.
+    - Symmetric CMK mainly used to wrap the DataKey, also can be used to encrypted an arbitrary set of bytes data(<6KB).
+    - Asymmetric CMK mainly used to sign/verify or asymmetric encrypt/decrypt datas (not for the DataKey.)
+ - DataKey is a symmetric key with random size of bytes.
+    - The user will use the plaintext of the DataKey to locally encrypt the data without using KMS and store the encrypted data together with the ciphertext of the DataKey, then clear the plaintext of DataKey from memory as soon as possible after use.
```

Comment:
The plaintext of the DataKey is used to encrypt data locally, which should be cleared from memory once been used as soon as possible.